core/server/utils.go

```diff
@@ -18,3 +18,11 @@ func GetTenant(namespace, clusterName string, clusterUserNamespaces map[string][
 
 	return ""
 }
+
+func ExtractValueFromMap(mapName map[string]string, key string) string {
```

Comment:
This feels like a bit of an anti-pattern, although I don't know where it is used. This would hide a key not being defined, which is probably something that should be handled on a case by case basis.

Reply:
I've renamed it and added proper docstring for it